lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -44,14 +46,14 @@ def storage_method
       def old_unique_for?
         connection do |conn|
           conn.watch(payload_hash)
-          pid = conn.get(payload_hash).to_i
-          if pid == 1 || (pid == 2 && item['at'])
+          pid = conn.get(payload_hash)
+          if pid && (pid != 'scheduled' || item['at'])
             conn.unwatch
             nil
           else
             conn.multi do
               if expires_at > 0
-                conn.setex(payload_hash, expires_at, item['jid'])
+                conn.setex(payload_hash, expires_at, item['at'] ? 'scheduled' : item['jid'])
```

Comment:
Yes, so we wouldn't normally need the `jid` in the schedule - because SidekiqUnique never unlocks `scheduled` only promotes to the enqueued lock -- the exception being deletion of jobs from the scheduled queue via. the sidekiq API after for example a Sidekiq restart that would have allowed the lock to expire (thus would then allow a scheduled job to delete the `scheduled` mutex of another job). 

Over all I'm not sure if it's worth caring that much about uniqueness in the scheduled queue, so I may submit a patch later to not have any kind of scheduled uniqueness by default -- but probably we should be as consistent as possible if it is going to be supported?